packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -232,6 +237,31 @@ Learn more about compiler configuration at https://hardhat.org/config"
   }
 }
 
+function checkUnsupportedRemappings(userConfig: any) {
```

Comment:
Maybe we can use HardhatUserConfig here, as it was already validated

Reply:
I decided not to simply so that this function remains in parity with the other function it's being called with (`checkMissingSolidityConfig`), but I'm happy to change that

Reply:
Makes sense